packages/next/build/swc/src/next_dynamic.rs

```diff
@@ -0,0 +1,174 @@
+                value: Box::new(Expr::Arrow(ArrowExpr {
+                  params: vec![],
+                  body: BlockStmtOrExpr::Expr(Box::new(Expr::Array(ArrayLit {
+                    elems: vec![Some(ExprOrSpread {
+                      expr: Box::new(Expr::Call(CallExpr {
+                        callee: ExprOrSuper::Expr(Box::new(Expr::Member(MemberExpr {
+                          obj: ExprOrSuper::Expr(Box::new(Expr::Ident(Ident {
+                            sym: js_word!("require"),
+                            span: DUMMY_SP,
+                            optional: false,
+                          }))),
+                          prop: Box::new(Expr::Ident(Ident {
+                            sym: "resolveWeak".into(),
+                            span: DUMMY_SP,
+                            optional: false,
+                          })),
+                          computed: false,
+                          span: DUMMY_SP,
+                        }))),
+                        args: vec![ExprOrSpread {
+                          expr: Box::new(Expr::Lit(Lit::Str(Str {
+                            value: self.filename.clone().to_string().into(),
+                            span: DUMMY_SP,
+                            kind: StrKind::Normal {
+                              contains_quote: false,
+                            },
+                            has_escape: false,
+                          }))),
+                          spread: None,
+                        }],
+                        span: DUMMY_SP,
+                        type_args: None,
+                      })),
+                      spread: None,
+                    })],
+                    span: DUMMY_SP,
+                  }))),
+                  is_async: false,
+                  is_generator: false,
+                  span: DUMMY_SP,
+                  return_type: None,
+                  type_params: None,
+                })),
+              }))),
+              PropOrSpread::Prop(Box::new(Prop::KeyValue(KeyValueProp {
+                key: PropName::Ident(Ident::new("modules".into(), DUMMY_SP)),
+                value: Box::new(Expr::Array(ArrayLit {
+                  elems: vec![Some(ExprOrSpread {
+                    expr: Box::new(Expr::Bin(BinExpr {
+                      span: DUMMY_SP,
+                      op: BinaryOp::Add,
+                      left: Box::new(Expr::Lit(Lit::Str(Str {
+                        value: file_from.into(),
+                        span: DUMMY_SP,
+                        kind: StrKind::Normal {
+                          contains_quote: false,
+                        },
+                        has_escape: false,
+                      }))),
+                      right: Box::new(Expr::Lit(Lit::Str(Str {
+                        value: import_specifier.unwrap(),
+                        span: DUMMY_SP,
+                        kind: StrKind::Normal {
```

Comment:
Note: this is correct usage of `Str.kind`, as this string is generated from an identifier.